lib/que/job.rb

```diff
@@ -77,6 +78,25 @@ def self.run(*args)
       new(args: args).tap { |job| job.run(*args) }
     end
 
+    def log_context_proc
+      # This method is redefined when custom_log_context is set on the class
+      nil
+    end
+
+    def self.custom_log_context(custom_proc)
+      if custom_proc.is_a?(Proc)
+        define_method(:log_context_proc) { custom_proc }
```

Comment:
Could we store the proc in a `class_attribute` instead?

Basically, something in this vein:

```rb
class_attribute :custom_log_context_proc

def self.custom_log_context(proc) do
  self.custom_log_context_proc = proc
end
```

And to retrieve the log context:

```rb
job_class.custom_log_context_proc&.call(job) || {}
```

Reply:
I'm happy with that @mattieuprat, see my update